### PR TITLE
[HUBS-1586] Updated ReadMe with new doc link

### PR DESCRIPTION
**Pull request by nick-mathison**
Made various updates to the ReadMe:

1) Updated documentation links
2) Various other edits to clarity. 

Note: I'm sure what's going on with the other go.mod, go.sum, and resource_vdb_test.go files. Let me know if they need to go.

### CODE REVIEW

**Comment by nick-mathison** [comment]
@sunnyx4 @arunskurian @kishlay-delphix 

Can you please help get this merged? Only the readme, not the other stuff.

File: README.md [hunk 1]
```diff
@@ -7,32 +7,32 @@
...
 
 ## Prerequisites
 
-1. Setup Delphix Control Tower(DCT) API Layer. For more information, visit [DCT Home.](https://docs.delphix.com/dct)
-2. Delphix Engines must be registered with DCT APIs
-3. API-Keys must be created for authenticating with DCT APIs. Refer to [DCT Home](https://docs.delphix.com/dct) for more info.
-4. Additional infrastructure required for testing the provider operations [ e.g Hosts to be added as environments, dSources to create VDBs from]
-5. Development setup for GoLang.
+1. Install Delphix Control Tower (DCT). For more information, visit the [DCT documentation](https://docs.delphix.com/dct).
+2. Delphix Continuous Data and Continuous Compliance engines must be connected to DCT.
```
Comment (kishlay-delphix) [inline README.md:15]: QQ: Do we provide any Continuous Compliance with our current terraform provider?

Comment (nick-mathison) [reply README.md:15]: Today, it's indirectly. 

For example, a VDB can have a post-hook that triggers a compliance job. But there's nothing today that will run a Job directly.

**Review by kishlay-delphix — APPROVED** [review]
Readme LGTM. Even though I'm not sure if we should check in the other files.

**Comment by nick-mathison** [comment]
Let me see if I pull them out.

**Comment by nick-mathison** [comment]
I think what happened is that I merged my branch with the main branch. So those 3 other file changes are in the main branch but were never added to develop. See Eyal's PR: https://github.com/delphix-integrations/terraform-provider-delphix/pull/36/files

We'll need to pull in those changes some way or another. Probably makes sense for you to rebase on your PR first, confirm it works, and then merge in my changes.

**Pull request closed without merging**